Makefile

```diff
@@ -1,50 +1,76 @@
+	docker build -t $(DOCKER_REMOTE_REPO)/$(APP_NAME)_nofeatures:$(VERSION) $(COMMON_DOCKER_PARAMS) --build-arg CARGO_PARAMS="--no-default-features" .
 
 run: ## Run container on port configured in `config.env`
 	docker run -it --rm --env HORUST_LOG=debug -v $(REPO_HOME)/examples/services/longrunning/:/etc/horust/services/ --name="$(NAME)" $(NAME):$(VERSION)
 
 run-bash: ## Run bash with horust
-	docker run -it --rm --env HORUST_LOG=debug --name="$(NAME)" $(DOCKER_REPO)/$(APP_NAME):$(VERSION) -- /bin/bash
+	docker run -it --rm --env HORUST_LOG=debug --name="$(NAME)" $(DOCKER_REMOTE_REPO)/$(APP_NAME):$(VERSION) -- /bin/bash
 
 stop: ## Stop and remove a running container
 	docker stop $(NAME)
 
-# Docker publish
+# Docker publishing tasks
 publish: build publish-latest publish-version ## publish the `{version}` ans `latest` tagged containers to ECR
 
 publish-latest: tag-latest ## publish the `latest` tagged container
-	@echo 'publish latest to $(DOCKER_REPO)'
-	docker push $(DOCKER_REPO)/$(APP_NAME):latest
+	@echo 'publish latest to $(DOCKER_REMOTE_REPO)'
+	docker push $(DOCKER_REMOTE_REPO)/$(APP_NAME):latest
 
 publish-version: ## publish the `{version}` tagged container to ECR
-	@echo 'publish $(VERSION) to $(DOCKER_REPO)'
-	docker push $(DOCKER_REPO)/$(APP_NAME):$(VERSION)
+	@echo 'publish $(VERSION) to $(DOCKER_REMOTE_REPO)'
+	docker push $(DOCKER_REMOTE_REPO)/$(APP_NAME):$(VERSION)
 
-tag-latest:
+tag-latest: ## tags the latest container with the version listed above
 	@echo 'create tag latest'
-	docker tag $(DOCKER_REPO)/$(APP_NAME):$(VERSION) $(DOCKER_REPO)/$(APP_NAME):latest
+	docker tag $(DOCKER_REMOTE_REPO)/$(APP_NAME):$(VERSION) $(DOCKER_REMOTE_REPO)/$(APP_NAME):latest
 
 version: ## output to version
-	@echo $(VERSION)
\ No newline at end of file
+	@echo ''$(VERSION)
+
+# Docker local development tasks
+## Dargo == Docker Cargo
+dargo-prep: ## This runs everything neccessary to start developing locally in a container
+	# Spin up a long-running rust container
+	make dargo-create-container
+	# Compile and cache all dependencies
+	make dargo COMMAND=build
+	# Perform a first-sweep check, to fill up the cache (it seems to be different than the build one)
+	make dargo COMMAND=check
+    # Compile and cache test dependencies, run tests
+	make dargo COMMAND=test
+	# Now all caches are filled, and subsequent operations will be faster
+	# Consider adding dargo(){ make dargo COMMAND=$1} to your ~/.zshrc or ~/.bashrc for ergonomics
+
+dargo: ## Run a cargo command inside the container
+	docker exec -ti docker-horust cargo $(COMMAND)
+
+dargo-create-container: ## Create a Rust container with this folder bind-mounted to it
+	@echo 'building rust image for local development'
+	docker build -t horust -f localdev/Dockerfile localdev/
+	@echo 'running interactive rust container for local development'
+	docker run -dt \
+ 	--name docker-horust \
+ 	--mount type=bind,source="$(shell pwd)",target=/usr/src/Horust \
```

Comment:
Is there a makefile command to get the makefile path? Because I would use that instead of pwd. Otherwise, please specify in the docs to CD to the horust home dir...

Reply:
I actually do specify it in the docs:)

https://github.com/FedericoPonzi/Horust/pull/41/files#diff-6a3371457528722a734f3c51d9238c13R99-R102

Reply:
I could use something like [this](https://stackoverflow.com/a/18137056/491214), but it feels a bit convoluted. I like `cd` into a place and do X Y Z better